subsys/mpsl/Kconfig

```diff
@@ -6,8 +6,31 @@
 
 menu "Nordic MPSL"
 
+config MPSL
```

Comment:
I'm not entirely convinced by this change. MPSL is stored in sdk-nrfxlib and so far the tree in menuconfig reflected that. MPSL showed under `Modules --> nrfxlib` with other libraries kept in there. AFAIU, this commit paves the way for moving MPSL Kconfig into `Modules --> nrf` branch of the tree. Is there any particular reason for doing that?

Reply:
one thing to remember is that when the Zephyr module is not available, the Kconfig related to that module should automatically be gone.

With `nrfxlib`, this becomes a philosophical discussion, because nrfxlib is always there, because a lot of things will break if it isn't. But the principle helps to decide where settings should be located.

Another view on this is maintenance.
If anyone changes something in `nrfxlib`, where should they find the right Kconfig to update ?
For example, when adding a new library variant that can be selected by the user.
For a simple library as `mpsl`, it may not be much of a problem, but think about security.
Not knowing what settings relates to a library can make it much harder to know where to update.

Today there are two locations related to Zephyr modules where the CMakeLists.txt and Kconfig files should go:
- Directly in the module itself: `nrfxlib/`
- In a modules ext root, for `nrf`-repo that is: [modules/nrfxlib/](https://github.com/nrfconnect/sdk-nrf/tree/main/modules/nrfxlib) (cmake-ext / kconfig-ext == True)

So what to do if wanting to provide a possibility for the user to also select the MPSL library under https://github.com/nrfconnect/sdk-nrf/blob/63170ce2f6147407b69390a0d3503ed11b6ecc88/subsys/mpsl/Kconfig

It is allowed to provide a setting twice in Kconfig.
Especially useful when changing `defaults`.
But it can also be used to add extra items to a choice.

In this case, one could have:
```
choice MPSL_BUILD_TYPE
	prompt "MPSL build type"

endchoice
```
in both places.
We do similar here: https://github.com/nrfconnect/sdk-nrf/blob/f9b4ba29078fe4947ce7537b03fdb77514d7c50b/Kconfig.nrf#L15-L19 where we extend an upstream choice menu.

The problem I see with the MPSL case, is that we do not have an item to put into the `choice MPSL_BUILD_TYPE` in the sdk-nrf repo, cause only one valid option exists today and that is the library.
We don't want to duplicate then choice item, only have it available at both locations.
But mayby someone has a good idea.

My current view is that we should keep the knowledge of the library in the nrfxlib, where the setting is used today, but having a selection choice available in `nrf/subsys/mpsl/` as a convenience to the user is fine, if we can do so in a clean way.

Reply:
in theory one can source the same Kconfig file again.

Problem with such solution is that it looses all higher level `depends on` that has been added with `if <SOMETHING>`, so such a solution can easily end-up with more issues than it solves, that is symbols suddenly being visible for configuration when they should not be visible.

Let's have a brief discussion on options.